packages/react/src/components/Tree/Tree.tsx

```diff
@@ -60,14 +83,45 @@ class Tree extends UIComponent<ReactProps<TreeProps>> {
     accessibility: defaultBehavior,
   }
 
+  static autoControlledProps = ['activeIndex']
+
+  getInitialAutoControlledState({ defaultActiveIndex, exclusive }): TreeState {
+    return {
+      activeIndex: defaultActiveIndex === undefined ? (exclusive ? -1 : []) : defaultActiveIndex,
+    }
+  }
+
+  computeNewIndex = (index: number) => {
+    const { activeIndex } = this.state
+    const { exclusive } = this.props
+
+    if (exclusive) return index
+    // check to see if index is in array, and remove it, if not then add it
+    return _.includes(activeIndex as number[], index)
+      ? _.without(activeIndex as number[], index)
+      : [...(activeIndex as number[]), index]
+  }
+
+  handleTreeItemOverrides = (predefinedProps: TreeItemProps) => ({
+    onTitleClick: (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+      this.trySetState({ activeIndex: this.computeNewIndex(treeItemProps.index) })
+      _.invoke(predefinedProps, 'onTitleClick', e, treeItemProps)
+    },
+  })
+
   renderContent() {
-    const { items, renderItemTitle } = this.props
+    const { items, renderItemTitle, exclusive } = this.props
+    const { activeIndex } = this.state
 
-    return _.map(items, item =>
+    return _.map(items, (item: ShorthandValue, index: number) =>
       TreeItem.create(item, {
         defaultProps: {
+          index,
+          exclusive,
           renderItemTitle,
+          open: exclusive ? index === activeIndex : _.includes(activeIndex as number[], index),
```

Comment:
Normalize the `activeIndex` as an array before doing this checks. If I don't use the exclusive, but use just defaultActiveIndex={3}, this may break. Do something like:
```
const activeIndexAsArray = _.isArray(activeIndex) ? activeIndex : [activeIndex]
```

and then always use the `_.include`

Reply:
This is consistent with the accordion.
<img width="763" alt="screen shot 2019-03-07 at 6 09 57 pm" src="https://user-images.githubusercontent.com/1810322/53957402-4bdd9c00-4104-11e9-8c94-d5be6039b32d.png">

Reply:
Still, the user may provide the example showed above. If this is how it is implemented the Accordion, it is another story and we may fix it there as well.